### PR TITLE
Add Atmel SAM0 pinctrl for SPI, I2C & TCC

### PR DESCRIPTION
**Pull request by attie-argentum**
This PR adds pinctrl support for the SAM0 SPI, I2C and TCC drivers, as a continuation of #35423, and in support of #41308.

### CODE REVIEW

**Comment by nandojve** [comment]
Hi @attie-argentum , you got the spirit !

**Comment by attie-argentum** [comment]
I'm unsure how we should handle `samr2x-pinctrl.dtsi`... there are a number of functions that the SAMD2x parts have available that are not available on the SAML2x parts - for example:
- On SAML2x, it looks like perhaps PA02 & PA03 aren't bonded out
- On SAML2x, PA04 has `TCC0/WO[0]` on mode `E`, but _does not_ have `TCC3/WO[2]` on mode `F` like the SAMD2x.

For pads that aren't bonded, I'm fairly happy to leave the pinctrl items in place (it's "_harmless_"?), but for modes that aren't available (e.g: `PA04.F`), I'd be tempted to use a `/delete-node/` in `samr2x-pinctrl.dtsi`... I suspect there could be many... any thoughts?

**Comment by attie-argentum** [comment]
I believe this is now ready for review.

I've tested this on SAML21 XPro and SAMR34 XPro boards (via #41308), as well as our custom board.

**Review by gmarull — CHANGES_REQUESTED** [review]
This PR is not really implementing a pinctrl driver but just moving `pinmux.c` to DT. Please check https://docs.zephyrproject.org/latest/guides/pinctrl/index.html and other existing implementations listed in https://github.com/zephyrproject-rtos/zephyr/issues/39740

**Comment by attie-argentum** [comment]
Hi @gmarull,

I'm not sure what you mean - this PR follows directly in the footsteps of https://github.com/zephyrproject-rtos/zephyr/pull/35423/commits/7c91f46f0584f21bb14eac62610594bc1ff2eadc (#35423, which is listed in #39740).

Please could you clarify?

**Comment by gmarull** [comment]
> Hi @gmarull,
> 
> I'm not sure what you mean - this PR follows directly in the footsteps of [7c91f46](https://github.com/zephyrproject-rtos/zephyr/commit/7c91f46f0584f21bb14eac62610594bc1ff2eadc) (from #35423, which is listed in #39740).
> 
> Please could you clarify?

I can't see any `drivers/pinctrl` driver here, nor any of the pinctrl APIs being used. Please check the references I've provided, this is using a custom API.

**Comment by attie-argentum** [comment]
Hi @gmarull, understood... If you're pushing for a "proper" pinctrl driver right now, then I'll have to put this PR on hold for the moment. Hopefully I'll get back to it, but it may be a couple of weeks.

**Comment by attie** [comment]
After discussion, this PR is on hold, and I'm going to try and spend some time on a proper pinctrl driver in the next month or so...

**Comment by attie-argentum** [comment]
After discussion, this PR is on hold, and I'm going to try and spend some time on a proper pinctrl driver in the next month or so...

File: dts/bindings/spi/atmel,sam0-spi.yaml [hunk 1]
```diff
@@ -27,6 +27,19 @@ properties:
       required: true
       description: Data Out Pinout
 
+    pinctrl-0:
+      type: phandles
+      description: |
+        PORT pin configuration for all SPI signals.  We expect that
+        the phandles will reference pinctrl nodes.  These nodes will
+        have a nodelabel that matches the Atmel SoC HAL defines and
+        be of the form p<port><pin><periph>_<inst>_<signal>.
```
Comment (Cooler1989) [inline dts/bindings/spi/atmel,sam0-spi.yaml:36]: p<port><pin>[missing '_']<periph>_<inst>_<signal>.

**Comment by nandojve** [comment]
I'm closing this in favor of #43731. New devices should have their pins defined as https://github.com/zephyrproject-rtos/hal_atmel/pull/19 pinconfigs/README and will be accept after #43731.

**Comment by attie-argentum** [comment]
Thanks @nandojve !

**Pull request closed without merging**